### PR TITLE
Translate `lit_tests/incrParse` with `assertIncrementalParse`

### PR DESCRIPTION
**Pull request by StevenWong12**
This should make it easier to read and run test cases in lit_tests/incrParse with `assertIncrementalParse`.

### CODE REVIEW

**Review by ahoppen — COMMENTED** [review]
Thanks. This is going in a great direction.

File: Tests/SwiftParserTest/IncrementalParsingTests.swift [hunk 1]
```diff
@@ -37,4 +37,335 @@ public class IncrementalParsingTests: XCTestCase {
...
+        func parensAdded⏩️⏸️()⏪️ {
+        }
+
+        func openingBraceAdded() <<ADD_OPENING_BRACE|||{>>>
+
+        func closingBraceAdded() {
+
+        ⏩️⏸️}⏪️
+
+        ⏩️func⏸️⏪️ funcKeywordRemoved() {
```
[the remark this answers is not in the document]
Comment (StevenWong12) [reply Tests/SwiftParserTest/IncrementalParsingTests.swift:83]: If we split these up, do you think should we maintain all other text in in that file? I'm not quite sure how much these context will affect our test.

[the remark this answers is not in the document]
Comment (ahoppen) [reply Tests/SwiftParserTest/IncrementalParsingTests.swift:83]: I would just keep the immediate parent contexts. E.g. here, I would just keep `class InvalidFuncDecls {`

**Comment by StevenWong12** [comment]
I've also opened a separated pr (#1807) to delete all the old incremental parse test utils and `lit`-based test cases.

File: Tests/SwiftParserTest/IncrementalParsingTests.swift [hunk 2]
```diff
@@ -37,4 +37,395 @@ public class IncrementalParsingTests: XCTestCase {
...
+      ]
+    )
+  }
+
+  public func testNextTokenCalculation() throws {
+    try XCTSkipIf(true, "Swift parser does not handle node reuse yet")
+    assertIncrementalParse(
+      """
+      let a = "hello"
+          let c = "⏩️ ⏸️⏪️world"
```
Comment (ahoppen) [inline Tests/SwiftParserTest/IncrementalParsingTests.swift:280]: I don’t recall why but in reuse.swift it has the following comment and `let a` is indented by four spaces. I think we should maintain that indentation and comment.

```swift
// The indentation on these lines is important for the test case
```

**Review by ahoppen — COMMENTED** [review]
One minor comment, otherwise LGTM.

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Pull request merged**